### PR TITLE
Fix npm 5

### PR DESCRIPTION
**Pull request by ILMTitan**
Change the build system to work with npm version 5 (the version that comes with NodeJs 8).
Also, update the vsts-task-lib to 2.1.0 from 1.1.0.

### CODE REVIEW

File: build/BuildFunctions.psm1 [hunk 1]
```diff
@@ -53,77 +40,30 @@ function Initialize-PsTask([string]$task) {
...
-        npm install ../common --loglevel warn | Out-Null
-
-        Write-Verbose "Running: tsc"
-        tsc
-        if ($LASTEXITCODE -ne 0) {
-            throw "tsc failed for task $task"
+    $allJobErrors = @()
+    do {
+        if ($jobs.Count -gt 1) {
+            $jobs | Wait-Job -Any | Receive-Job -Wait -AutoRemoveJob -ErrorVariable jobErrors
```
Comment (quoctruong) [inline build/BuildFunctions.psm1:57]: Do we need `Wait-Job` if we already have `Receive-Job -Wait`?

File: build/BuildFunctions.psm1 [hunk 2]
```diff
@@ -53,77 +40,30 @@ function Initialize-PsTask([string]$task) {
...
-    try {
-        if (-not (Test-Path node_modules)) {
-            npm install --loglevel error | Out-Null
-        }
-    } finally {
-        popd
-    }
-}
+function Invoke-AllMochaTests([string[]]$tasks, [string]$reporter, [switch]$throwOnError) {
+    Write-Host "Testing Tasks"
```
Comment (quoctruong) [inline build/BuildFunctions.psm1:44]: Should we put `$tasks` in here so we know which ones we're testing?

File: build/BuildFunctions.psm1 (hunk 1 above)
Comment (ILMTitan) [reply build/BuildFunctions.psm1:57]: `Receive-Job -Wait` will output to the console the output of all jobs, interlacing the output. `Wait-Job -Any` will return when a single job is complete. This combination allows us to display test results to the console as quickly as they are available, but in single understandable chunks.

File: build/BuildFunctions.psm1 (hunk 2 above)
Comment (ILMTitan) [reply build/BuildFunctions.psm1:44]: I don't think it is necessary. Each `Invoke-MochaTest` invocation says which test is being run already.

**Review by quoctruong — APPROVED** [review]

**Pull request merged**